enhancements/storage/csi-snapshot.md

```diff
@@ -37,52 +38,98 @@ We want to include the upstream CSI Snapshot feature in OpenShift.
+* Enable snapshot feature by default and provide the same support as upstream 1.17 release.
 
-* Package and ship a downstream image of the this sidecar.
+* Release and maintain downstream csi-external-snapshotter, csi-external-provisioner sidecars images off of the upstream images based on Kubernetes 1.17 to allow Red Hat CSI drivers (such as OCS) to consume them.
 
 ### Non-Goals
 
-## Proposal
+## Upstream status
+This feature has reached v1beta status upstream and is described in https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20190709-csi-snapshot.md and https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/csi-snapshot.md.
+
+This feature poses new requirements on Kubernetes distributions such as OpenShift:
+* Ship and install snapshot CRDs.
+* Ship and run snapshot-controller in a cluster.
+  * snapshot-controller is very similar to in-tree PV controller in its function, it watches VolumeSnapshots and provisions VolumeSnapshotContents for them. The snapshot controller is independent on storage backend and only one runs for all CSI drivers installed in a cluster.
+
+Upstream has chosen [addon](https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/volumesnapshots) for both. Upstream uses [StatefulSet for the snapshot-controller with single replica](https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/volumesnapshots/volume-snapshot-controller/volume-snapshot-controller-deployment.yaml) for the snapshot-controller.
 
-This feature has already been implemented upstream, and is described in https://github.com/kubernetes/enhancements/blob/master/keps/sig-storage/20190709-csi-snapshot.md . It is currently documented upstream at https://kubernetes.io/docs/concepts/storage/volume-snapshots/ .
+## Proposal
 
-We will be releasing the `external-snapshotter` sidecar image to enable CSI Snapshot support in OCP as Developer Preview only. This sidecar can be used by internal teams, such as CNV or OCS, for their drivers.
+1. OCP ships a new operator for csi-snapshot-controller, say csi-snapshot-controller-operator.
+  * This operator will be installed by CVO in all clusters.
+    * We do not know in advance what CSI drivers will a cluster admin install and we want the cluster ready for snapshots.
+  * This operator will create / update VolumeSnapshot, VolumeSnapshotContent and VolumeSnapshotClass CRDs.
+    * Vanilla copy of [upstream CRDs](https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/volumesnapshots/crd) is used.
+  * This operator will create Deployment with csi-snapshot-controller
+    * Using Deployment with 3 replicas + leader election instead of upstream StatefulSet - leader election is significantly faster to run a new leader when a node with the current leader gets unavailable.
+    * Optionally, run the Deployment on masters using proper node selector + tolerations.
```

Comment:
This doesn't need to be in a first cut.  The idea that masters are generally special doesn't hold up without strong assumptions on topology that we abandoned in 4.x generally.